Terraform plugin crashing v 1.7.0

**Description**
Error: The terraform-provider-rollbar_v1.7.0 plugin crashed constantly and does not recover

**Context**
The user had a good deployment yesterday; however, they ran a plan today to check the build's status. They're seeing the Rollbar provider crash now consistently. They reported having seen this before and had to tear down and rebuild our Terraform project from the ground up a few times.

The user is to have the notifications built from a standard module that all projects can reference to ensure consistency in notifications over email and slack.

**Stack trace** 
terraform-provider-rollbar_v1.7.0 plugin

panic: Invalid address to set: []string{"config", "1987373520", "summary_time"}

goroutine 46 [running]:
github.com/rollbar/terraform-provider-rollbar/rollbar.mustSet(...)
github.com/rollbar/terraform-provider-rollbar/rollbar/provider.go:117
github.com/rollbar/terraform-provider-rollbar/rollbar.resourceNotificationRead({0xd0f498, 0xc00028da40}, 0x7f7edf17ba68, {0xb2be80, 0xc000457ce0})
github.com/rollbar/terraform-provider-rollbar/rollbar/resource_notification.go:345 +0x615
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00030ca80, {0xd0f498, 0xc00028da40}, 0x24, {0xb2be80, 0xc000457ce0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00030ca80, {0xd0f498, 0xc00028da40}, 0xc00015b450, {0xb2be80, 0xc000457ce0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00030e750, {0xd0f498, 0xc00028da40}, 0xc00028dac0)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000338c80, {0xd0f540, 0xc0002d9cb0}, 0xc000281560)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x3b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xbce1a0, 0xc000338c80}, {0xd0f540, 0xc0002d9cb0}, 0xc000281500, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000324700, {0xd1c678, 0xc000464300}, 0xc00037c300, 0xc000456ba0, 0x11fae50, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1217 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc000324700, {0xd1c678, 0xc000464300}, 0xc00037c300, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1540 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.36.0/server.go:878 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.36.0/server.go:876 +0x294
panic: Invalid address to set: []string{"config", "1987373520", "min_item_level"}

goroutine 120 [running]:
github.com/rollbar/terraform-provider-rollbar/rollbar.mustSet(...)
github.com/rollbar/terraform-provider-rollbar/rollbar/provider.go:117
github.com/rollbar/terraform-provider-rollbar/rollbar.resourceNotificationRead({0xd0f498, 0xc00052a680}, 0x7f06b0704108, {0xb2be80, 0xc00055eae0})
github.com/rollbar/terraform-provider-rollbar/rollbar/resource_notification.go:345 +0x615
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00012ca80, {0xd0f498, 0xc00052a680}, 0x24, {0xb2be80, 0xc00055eae0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00012ca80, {0xd0f498, 0xc00052a680}, 0xc00079c9c0, {0xb2be80, 0xc00055eae0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00012e750, {0xd0f498, 0xc00052a680}, 0xc00052a800)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000156c80, {0xd0f540, 0xc000283020}, 0xc000758f00)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x3b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xbce1a0, 0xc000156c80}, {0xd0f540, 0xc000283020}, 0xc000758ea0, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000142700, {0xd1c678, 0xc00047e300}, 0xc000294200, 0xc00046ed50, 0x11fae50, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1217 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc000142700, {0xd1c678, 0xc00047e300}, 0xc000294200, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1540 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.36.0/server.go:878 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.36.0/server.go:876 +0x294
panic: Invalid address to set: []string{"config", "1987373520", "summary_time"}

goroutine 115 [running]:
github.com/rollbar/terraform-provider-rollbar/rollbar.mustSet(...)
github.com/rollbar/terraform-provider-rollbar/rollbar/provider.go:117
github.com/rollbar/terraform-provider-rollbar/rollbar.resourceNotificationRead({0xd0f498, 0xc0008241c0}, 0x7fe671104a68, {0xb2be80, 0xc0004b11d0})
github.com/rollbar/terraform-provider-rollbar/rollbar/resource_notification.go:345 +0x615
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0002a2a80, {0xd0f498, 0xc0008241c0}, 0x24, {0xb2be80, 0xc0004b11d0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:358 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0002a2a80, {0xd0f498, 0xc0008241c0}, 0xc000a16750, {0xb2be80, 0xc0004b11d0})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00029a768, {0xd0f498, 0xc0008241c0}, 0xc000a0c000)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00029cd00, {0xd0f540, 0xc00058dfb0}, 0xc000431aa0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:553 +0x3b0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xbce1a0, 0xc00029cd00}, {0xd0f540, 0xc00058dfb0}, 0xc000431a40, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002b4700, {0xd1c678, 0xc00046e300}, 0xc0005f2100, 0xc00045ed20, 0x11fae50, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1217 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc0002b4700, {0xd1c678, 0xc00046e300}, 0xc0005f2100, 0x0)
google.golang.org/grpc@v1.36.0/server.go:1540 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.36.0/server.go:878 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.36.0/server.go:876 +0x294

**Suggestion**

It would be beneficial to report the crash to the plugin's maintainers so it
can be fixed. The output above should help diagnose the issue.

--- a/client/notification.go
+++ b/client/notification.go
@@ -23,9 +23,10 @@
 package client
 
 import (
-	"github.com/rs/zerolog/log"
 	"strconv"
 	"strings"
+
+	"github.com/rs/zerolog/log"
 )
 
 type Notification struct {
@@ -163,6 +164,39 @@ func (c *RollbarAPIClient) DeleteNotification(notificationID int, channel string
 	}
 	l.Debug().Msg("Notifications successfully deleted")
 	return nil
+}
+
+func (c *RollbarAPIClient) ListNotifications(channel string) ([]Notification, error) {
+	u := c.BaseURL + pathNotificationCreate
+
+	l := log.With().
+		Logger()
+	l.Debug().Msg("Reading notifications from API")
+
+	resp, err := c.Resty.R().
+		SetResult(notificationsResponse{}).
+		SetError(ErrorResult{}).
+		SetPathParams(map[string]string{
+			"channel": channel,
+		}).
+		Get(u)
+
+	if err != nil {
+		l.Err(err).Msg(resp.Status())
+		return nil, err
+	}
+	err = errorFromResponse(resp)
+	if err != nil {
+		l.Err(err).Send()
+		return nil, err
+	}
+	nr := resp.Result().(*notificationsResponse)
+	if nr.Err != 0 {
+		l.Warn().Msg("Notification not found")
+		return nil, ErrNotFound
+	}
+	l.Debug().Msg("Notification successfully read")
+	return nr.Result, nil
 }
 
 type notificationResponse struct {

--- a/client/service_link.go
+++ b/client/service_link.go
@@ -23,8 +23,9 @@
 package client
 
 import (
-	"github.com/rs/zerolog/log"
 	"strconv"
+
+	"github.com/rs/zerolog/log"
 )
 
 type ServiceLink struct {
@@ -158,7 +159,41 @@ func (c *RollbarAPIClient) DeleteServiceLink(id int) error {
 	return nil
 }
 
+func (c *RollbarAPIClient) ListSerivceLinks() ([]ServiceLink, error) {
+	u := c.BaseURL + pathServiceLinkCreate
+
+	l := log.With().
+		Logger()
+	l.Debug().Msg("Reading service links from API")
+
+	resp, err := c.Resty.R().
+		SetResult(serviceLinksResponse{}).
+		SetError(ErrorResult{}).
+		Get(u)
+
+	if err != nil {
+		l.Err(err).Msg(resp.Status())
+		return nil, err
+	}
+	err = errorFromResponse(resp)
+	if err != nil {
+		l.Err(err).Send()
+		return nil, err
+	}
+	sl := resp.Result().(*serviceLinksResponse)
+	if sl.Err != 0 {
+		l.Warn().Msg("Service link not found")
+		return nil, ErrNotFound
+	}
+	l.Debug().Msg("Service link successfully read")
+	return sl.Result, nil
+}
+
 type serviceLinkResponse struct {
 	Err    int         `json:"err"`
 	Result ServiceLink `json:"result"`
+}
+type serviceLinksResponse struct {
+	Err    int           `json:"err"`
+	Result []ServiceLink `json:"result"`
 }

--- a/rollbar/resource_integration_test.go
+++ b/rollbar/resource_integration_test.go
@@ -1,0 +1,101 @@
+/*
+ * Copyright (c) 2022 Rollbar, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package rollbar
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// TestIntegrationCreate tests creating an integration
+func (s *AccSuite) TestIntegrationCreate() {
+	integrationResourceName := "rollbar_integration.webhook_integration"
+	// language=hcl
+	config := `
+		resource "rollbar_integration" "webhook_integration" {
+  webhook {
+    enabled = true
+    url     = "https://www.rollbar.com"
+  }
+}
+	`
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(integrationResourceName),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.enabled", "true"),
+				),
+			},
+		},
+	})
+}
+
+// TestIntegrationUpdate tests updating an integration
+func (s *AccSuite) TestIntegrationUpdate() {
+	integrationResourceName := "rollbar_integration.webhook_integration"
+	// language=hcl
+	config1 := `
+	resource "rollbar_integration" "webhook_integration" {
+  webhook {
+    enabled = true
+    url     = "https://www.rollbar.com"
+  }
+}
+	`
+	config2 := `
+		resource "rollbar_integration" "webhook_integration" {
+  webhook {
+    enabled = false
+    url     = "https://www.rollbar.com"
+  }
+}
+	`
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config1,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(integrationResourceName),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.enabled", "true"),
+				),
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(integrationResourceName),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(integrationResourceName, "webhook.0.enabled", "false"),
+				),
+			},
+		},
+	})
+}

--- a/rollbar/resource_notification.go
+++ b/rollbar/resource_notification.go
@@ -36,7 +36,8 @@ import (
 
 var configMap = map[string][]string{"email": {"users", "teams"},
 	"slack":     {"message_template", "channel", "show_message_buttons"},
-	"pagerduty": {"service_key"}}
+	"pagerduty": {"service_key"},
+	"webhook":   {"url", "format"}}
 
 func CustomNotificationImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
 	splitID := strings.Split(d.Id(), ComplexImportSeparator)
@@ -281,14 +282,11 @@ func resourceNotificationUpdate(ctx context.Context, d *schema.ResourceData, m i
 
 func flattenConfig(config map[string]interface{}) *schema.Set {
 	var out = make([]interface{}, 0)
-	m := make(map[string]interface{})
-	for key, value := range config {
-		m[key] = value
-		out = append(out, m)
-	}
+	out = append(out, config)
 	specResource := resourceNotification().Schema["config"].Elem.(*schema.Resource)
 	f := schema.HashResource(specResource)
-	return schema.NewSet(f, out)
+	g := schema.NewSet(f, out)
+	return g
 }
 
 func flattenRule(filters []interface{}, trigger string) *schema.Set {
@@ -317,10 +315,10 @@ func flattenRule(filters []interface{}, trigger string) *schema.Set {
 	m["filters"] = filters
 	out = append(out, m)
 	m["trigger"] = trigger
-	out = append(out, m)
 	specResource := resourceNotification().Schema["rule"].Elem.(*schema.Resource)
 	f := schema.HashResource(specResource)
-	return schema.NewSet(f, out)
+	g := schema.NewSet(f, out)
+	return g
 }
 
 func resourceNotificationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {

--- a/rollbar/resource_notification.go
+++ b/rollbar/resource_notification.go
@@ -285,8 +285,8 @@ func flattenConfig(config map[string]interface{}) *schema.Set {
 	out = append(out, config)
 	specResource := resourceNotification().Schema["config"].Elem.(*schema.Resource)
 	f := schema.HashResource(specResource)
-	g := schema.NewSet(f, out)
-	return g
+	set := schema.NewSet(f, out)
+	return set
 }
 
 func flattenRule(filters []interface{}, trigger string) *schema.Set {
@@ -317,8 +317,8 @@ func flattenRule(filters []interface{}, trigger string) *schema.Set {
 	m["trigger"] = trigger
 	specResource := resourceNotification().Schema["rule"].Elem.(*schema.Resource)
 	f := schema.HashResource(specResource)
-	g := schema.NewSet(f, out)
-	return g
+	set := schema.NewSet(f, out)
+	return set
 }
 
 func resourceNotificationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {

--- a/rollbar/resource_notification_test.go
+++ b/rollbar/resource_notification_test.go
@@ -1,0 +1,186 @@
+/*
+ * Copyright (c) 2022 Rollbar, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package rollbar
+
+import (
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/rollbar/terraform-provider-rollbar/client"
+	"github.com/rs/zerolog/log"
+)
+
+func init() {
+	resource.AddTestSweepers("rollbar_notification", &resource.Sweeper{
+		Name: "rollbar_notification",
+		F:    sweepResourceNotification,
+	})
+}
+
+// TestNotificationCreate tests creating a project access token.
+func (s *AccSuite) TestNotificationCreate() {
+	notificationResourceName := "rollbar_notification.webhook_notification"
+	// language=hcl
+	config := `
+		resource "rollbar_notification" "webhook_notification" {
+  rule  {
+    filters {
+        type =  "environment"
+        operation =  "eq"
+        value = "production"
+    }
+    filters {
+       type = "framework"
+       operation = "eq"
+       value = 13
+    }
+   trigger = "new_item"
+  }
+  channel = "webhook"
+  config  {
+     url = "https://www.rollbar.com"
+     format = "json"
+  }
+}
+	`
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(notificationResourceName),
+					resource.TestCheckResourceAttr(notificationResourceName, "channel", "webhook"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.0.type", "environment"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.1.type", "framework"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.trigger", "new_item"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.format", "json"),
+				),
+			},
+		},
+	})
+}
+
+// TestNotificationCreate tests creating a project access token.
+func (s *AccSuite) TestNotificationUpdate() {
+	notificationResourceName := "rollbar_notification.webhook_notification"
+	// language=hcl
+	config1 := `
+		resource "rollbar_notification" "webhook_notification" {
+  rule  {
+    filters {
+        type =  "environment"
+        operation =  "eq"
+        value = "production"
+    }
+    filters {
+       type = "framework"
+       operation = "eq"
+       value = 13
+    }
+   trigger = "new_item"
+  }
+  channel = "webhook"
+  config  {
+     url = "https://www.rollbar.com"
+     format = "json"
+  }
+}
+	`
+	config2 := `
+		resource "rollbar_notification" "webhook_notification" {
+  rule  {
+    filters {
+        type =  "environment"
+        operation =  "eq"
+        value = "production"
+    }
+    filters {
+       type = "framework"
+       operation = "eq"
+       value = 13
+    }
+   trigger = "new_item"
+  }
+  channel = "webhook"
+  config  {
+     url = "https://www.rollbar.com"
+     format = "xml"
+  }
+}
+	`
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config1,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(notificationResourceName),
+					resource.TestCheckResourceAttr(notificationResourceName, "channel", "webhook"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.0.type", "environment"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.1.type", "framework"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.trigger", "new_item"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.format", "json"),
+				),
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(notificationResourceName),
+					resource.TestCheckResourceAttr(notificationResourceName, "channel", "webhook"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.0.type", "environment"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.filters.1.type", "framework"),
+					resource.TestCheckResourceAttr(notificationResourceName, "rule.0.trigger", "new_item"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.url", "https://www.rollbar.com"),
+					resource.TestCheckResourceAttr(notificationResourceName, "config.0.format", "xml"),
+				),
+			},
+		},
+	})
+}
+
+// sweepResourceUser cleans up orphaned Rollbar users.
+func sweepResourceNotification(_ string) error {
+	log.Info().Msg("Cleaning up Rollbar users from acceptance test runs.")
+
+	c := client.NewClient(client.DefaultBaseURL, os.Getenv("ROLLBAR_PROJECT_API_KEY"))
+	notifications, err := c.ListNotifications("webhook")
+	if err != nil {
+		log.Err(err).Send()
+		return err
+	}
+	for _, n := range notifications {
+		err = c.DeleteProject(n.ID)
+		if err != nil {
+			log.Err(err).Send()
+			return err
+		}
+	}
+	return nil
+}

--- a/rollbar/resource_notification_test.go
+++ b/rollbar/resource_notification_test.go
@@ -37,7 +37,7 @@ func init() {
 	})
 }
 
-// TestNotificationCreate tests creating a project access token.
+// TestNotificationCreate tests creating a notification
 func (s *AccSuite) TestNotificationCreate() {
 	notificationResourceName := "rollbar_notification.webhook_notification"
 	// language=hcl
@@ -84,7 +84,7 @@ func (s *AccSuite) TestNotificationCreate() {
 	})
 }
 
-// TestNotificationCreate tests creating a project access token.
+// TestNotificationUpdate tests updating a notification
 func (s *AccSuite) TestNotificationUpdate() {
 	notificationResourceName := "rollbar_notification.webhook_notification"
 	// language=hcl
@@ -165,9 +165,9 @@ func (s *AccSuite) TestNotificationUpdate() {
 	})
 }
 
-// sweepResourceUser cleans up orphaned Rollbar users.
+// sweepResourceNotification cleans up notifications
 func sweepResourceNotification(_ string) error {
-	log.Info().Msg("Cleaning up Rollbar users from acceptance test runs.")
+	log.Info().Msg("Cleaning up Rollbar notifications from acceptance test runs.")
 
 	c := client.NewClient(client.DefaultBaseURL, os.Getenv("ROLLBAR_PROJECT_API_KEY"))
 	notifications, err := c.ListNotifications("webhook")
@@ -176,7 +176,7 @@ func sweepResourceNotification(_ string) error {
 		return err
 	}
 	for _, n := range notifications {
-		err = c.DeleteProject(n.ID)
+		err = c.DeleteNotification(n.ID, "webhook")
 		if err != nil {
 			log.Err(err).Send()
 			return err

--- a/rollbar/resource_service_link_test.go
+++ b/rollbar/resource_service_link_test.go
@@ -108,8 +108,6 @@ func (s *AccSuite) TestServiceLinkUpdate() {
 			},
 		},
 	})
-	c := s.provider.Meta().(map[string]*client.RollbarAPIClient)[projectKeyToken]
-	c.ListSerivceLinks()
 }
 
 // sweepResourceServiceLink cleans up service links.

--- a/rollbar/resource_service_link_test.go
+++ b/rollbar/resource_service_link_test.go
@@ -1,0 +1,133 @@
+/*
+ * Copyright (c) 2022 Rollbar, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package rollbar
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/rollbar/terraform-provider-rollbar/client"
+	"github.com/rs/zerolog/log"
+)
+
+func init() {
+	resource.AddTestSweepers("rollbar_service_link", &resource.Sweeper{
+		Name: "rollbar_service_link",
+		F:    sweepResourceNotification,
+	})
+}
+
+// TestServiceLinkCreate tests creating a service link
+func (s *AccSuite) TestServiceLinkCreate() {
+	serviceLinkResourceName := "rollbar_service_link.service_link"
+	// language=hcl
+	tmpl := `
+		resource "rollbar_service_link" "service_link" {
+          name = "%s"
+     	  template = "sometemplate_new.{{ss}}"
+       }
+	`
+
+	config := fmt.Sprintf(tmpl, s.randName)
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(serviceLinkResourceName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "name", s.randName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "template", "sometemplate_new.{{ss}}"),
+				),
+			},
+		},
+	})
+}
+
+// TestServiceLinkUpdate tests updating a service link
+func (s *AccSuite) TestServiceLinkUpdate() {
+	serviceLinkResourceName := "rollbar_service_link.service_link"
+	// language=hcl
+	tmpl1 := `
+		resource "rollbar_service_link" "service_link" {
+          name = "%s"
+     	  template = "sometemplate.{{ss}}"
+       }
+	`
+	config1 := fmt.Sprintf(tmpl1, s.randName)
+	tmpl2 := `
+		resource "rollbar_service_link" "service_link" {
+          name = "%s"
+     	  template = "sometemplate_new.{{ss}}"
+       }
+	`
+	config2 := fmt.Sprintf(tmpl2, s.randName)
+	resource.ParallelTest(s.T(), resource.TestCase{
+		PreCheck:     func() { s.preCheck() },
+		Providers:    s.providers,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config1,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(serviceLinkResourceName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "name", s.randName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "template", "sometemplate.{{ss}}"),
+				),
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					s.checkResourceStateSanity(serviceLinkResourceName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "name", s.randName),
+					resource.TestCheckResourceAttr(serviceLinkResourceName, "template", "sometemplate_new.{{ss}}"),
+				),
+			},
+		},
+	})
+	c := s.provider.Meta().(map[string]*client.RollbarAPIClient)[projectKeyToken]
+	c.ListSerivceLinks()
+}
+
+// sweepResourceServiceLink cleans up service links.
+func sweepResourceServiceLink(_ string) error {
+	log.Info().Msg("Cleaning up Rollbar service links from acceptance test runs.")
+
+	c := client.NewClient(client.DefaultBaseURL, os.Getenv("ROLLBAR_PROJECT_API_KEY"))
+	serviceLinks, err := c.ListSerivceLinks()
+	if err != nil {
+		log.Err(err).Send()
+		return err
+	}
+	for _, s := range serviceLinks {
+		err = c.DeleteServiceLink(s.ID)
+		if err != nil {
+			log.Err(err).Send()
+			return err
+		}
+	}
+	return nil
+}

